WordPress/gutenberg
# Try: "No results" block container for the query block

## Description

Adds a container block in which you can place any other blocks, that are displayed when the query (loop) has no results.

See https://github.com/WordPress/gutenberg/issues/36609, https://github.com/WordPress/gutenberg/issues/27302

This is a simplified version that:
- Does not make any changes to the post template which already displays a "no results" message in the editor but not the front.
- Does not implement the suggestion to have a setting in the query block that opens a modal with editable content.
- Does not output a default message.

Alternative: https://github.com/WordPress/gutenberg/pull/36225

Challenges:
How to explain how the block can be used in the query loop.
Test the many different scenarios (for example, if the query is set to only include sticky posts, and the install has no sticky posts, both the non-sticky posts and the "no results" message is shown, and this is a bug).

## Testing Instructions





1. Apply the PR
2. Activate a theme with support for the site editor
2. Trash all posts on your test install
3. In the site editor, add a query block to the home or index template (depending on what shows the blog in your active theme)
4. Add a "no results" block inside the query.
5. Add some blocks inside the no results block.
6. Save and confirm that the inner blocks are displaying correctly on the front.

## Screenshots 

## Types of changes




New feature

## Checklist:
